### DISCUSSION TITLE
如何学习好wpf

### DISCUSSION THREAD

**Discussion by MiaoWei814** [BYJRK/DotNet-Discussions#170] [WPF 开发]
本人刚入坑c# 学习wpf, 我翻来覆去始终没找到那种一系列的视频讲解wpf的从0到1 然后mvvm也是 然后还有ui框架HandyControl我都不知道该如何学习 我朋友建议我去油管上学习 我看了一抹多 始终感觉怪怪的 唉 跪求各位大佬看到次贴还望指点一二 非常感谢

**Comment by MiaoWei814** [comment]
如果有建议学习的视频地址 还不望贴一贴

**Reply by scarletNnn** [reply]
UP之前视频有提到的AngleSix、Kevin Bost的都不错，还有[TimCorey](https://www.youtube.com/@IAmTimCorey)，B站上就有刘铁猛的WPF和C#视频

**Comment by asdf130162** [comment]
视频涵盖不到所有的0到1，很多时候是收集碎片知识然后自己系统化知识体系

GitHub很多是炫技的项目新人上来啃不动的，新手不建议直接从HandyControl上手，项目结构设计都够你喝一壶的

MVVM框架从MVVMLight学起吧，跟着博客走一遍https://www.cnblogs.com/wzh2010/p/6920706.html

UI方面按我说的来：自己找博客照猫画虎学（其实这些你买一本WPF编程宝典就差不多了，粗略过一下，用到了查）
1、自定义样式
2、控件模板
3、数据模板
4、触发器、转换器
5、自定义用户控件，依赖属性
6、数据绑定、相对绑定、静态动态资源绑定
7、自定义样式库、转换器、icon 资源，动态切换ResourceDictionary，实现主题换肤、多语言
8、页面切换的导航器
9、事件聚合器

把上面的学了，基本UI 90%左右的普通业务都可以解决

想再深入学MVVM 从 Stylet入手https://github.com/canton7/Stylet/wiki
Stylet有很多IOC，ViewModel管理器这种更深入的玩法
不要上来就学Prism这种，太重量级了，而且很多人自己都没玩明白就教人用
先学会MVVMLight，再学Stylet就很快了

: )
